src/lib/IP.coffee

```diff
@@ -0,0 +1,38 @@
+module.exports = class IP
+  @types: [
+    'data'
+    'openBracket'
+    'closeBracket'
+  ]
+
+  # Creates as new IP object
+  # Valid types: 'data', 'openBracket', 'closeBracket'
+  constructor: (@type = 'data', @data = null, options = {}) ->
+    @groups = [] # sync groups
+    @scope = null # sync scope id
+    @stream = null # substream id
+    @owner = null # packet owner process
+    for key, val of options
+      this[key] = val
+
+  # Creates a new IP copying its contents by value not reference
+  clone: ->
```

Comment:
Fine by me. Note that currently packets are not cloned/dropped by default and are passed by reference instead. Cloning behavior is something that I thought should be enabled explicitly, but `clonable` would come useful in either case.